src/orders/views/OrderDetails/OrderNormalDetails/index.tsx

```diff
@@ -114,11 +114,37 @@ export const OrderNormalDetails: React.FC<OrderNormalDetailsProps> = ({
   const [fulfillmentWarehouse, setFulfillmentWarehouse] = React.useState<
     Warehouse
   >(null);
+
   React.useEffect(() => {
-    // @TODO this is wip
-    // exact logic for determining default
-    // warehouse will be added in future PR
-    setFulfillmentWarehouse(warehouses?.[0]);
+    const warehousesAvailability = warehouses?.map(warehouse => {
+      if (!order?.lines) {
+        return undefined;
+      }
```

Comment:
Since we check on the line 142 if `order?.lines` exists I think this could be removed

Reply:
It has to stay. Further down there is `forEach` method which breaks down when `lines` are undefined (happens when creating new orders). Cannot use optional chaining there due to linter issue (no-unused-expressions)